src/App.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', () => {
+      this.setState({ hasClock: false });
+    });
```

Comment:
in my opinion  you need  to add `event.preventDefault `according to conditions :
#hide the `Clock` on a right mouse click in the `document` (`contextmenu` event):
    ```
```js
    document.addEventListener('contextmenu', (event) => {
      event.preventDefault(); // not to show the context menu

      // put your code here
    });
```

Reply:
It is a condition from the task
- `the Clock should be visible only when the hasClock is true;`